### DISCUSSION TITLE
Enhancing Release Note Generation through Automation

### DISCUSSION THREAD

**Discussion by DetectiveCode** [bentoml/BentoML#4927] [Ideas]
Hello,

We are a group of researchers developing a new automated release note generation tool. Our tool is designed to be highly configurable and widely applicable, working seamlessly with many programming languages and without imposing restrictions or requirements such as commit conventions or workflow adjustments. Our goal is to automatically produce clear, concise, and relevant release notes personalised to your project’s specific needs.

Our approach combines configurability with advanced AI techniques. Specifically, we use supervised learning for categorising and prioritising commits, and large language models for summarising and personalising the content of release notes.

The tool is designed to work out of the box with minimal configuration. We are also actively working on features like interactive editing, seamless CI pipeline integration, and more. Our aim is to contribute this tool to the open-source community, saving time for maintainers and providing useful release note for users.

If you’re interested in our work, we would greatly appreciate 5-10 minutes of your time to provide invaluable feedback via our brief questionnaire: 👉🏻 [Automatic Release Note Generation Study](https://forms.gle/cjboQrBn1JexPAZs8) 👈🏻.

Below, you’ll find an example of a release note generated for this project using an early demo version of our tool.

Thank you for your time and consideration.

Best regards,
Release Notes Research Team

![screencapture-github-osslab-pku-SmartDraft-blob-main-evaluation-Libraries-And-Frameworks-bentoml-1-smartdraft-rn-automatic-md-2024-08-22-16_54_13](https://github.com/user-attachments/assets/84900f39-9397-49e3-b8f1-8d1c141de868)

**Comment by aarnphm** [comment]
nice description
